api/v1/server/handlers/monitoring/probe.go

```diff
@@ -0,0 +1,317 @@
+)
+
+func (m *MonitoringService) MonitoringPostRunProbe(ctx echo.Context, request gen.MonitoringPostRunProbeRequestObject) (gen.MonitoringPostRunProbeResponseObject, error) {
+	if !m.enabled {
+		return nil, fmt.Errorf("monitoring is not enabled")
+	}
+
+	tenant := ctx.Get("tenant").(*db.TenantModel)
+
+	if tenant == nil || tenant.ID == "" {
+		return nil, fmt.Errorf("tenant is required")
+	}
+
+	if !slices.Contains[[]string](m.permittedTenants, tenant.ID) {
+
+		err := fmt.Errorf("tenant is not a monitoring tenant for this instance")
+
+		if len(m.permittedTenants) > 0 {
+			m.l.Error().Err(err).Msgf("monitoring tenants are %v", m.permittedTenants)
+		} else {
+			m.l.Error().Err(err).Msg("no monitoring tenants are configured")
+		}
+
+		return gen.MonitoringPostRunProbe403JSONResponse{}, err
+	}
+
+	token, err := getBearerTokenFromRequest(ctx.Request())
+
+	if err != nil {
+		return nil, err
+	}
+
+	events := make(chan string, 50)
+	errorChan := make(chan error, 50)
+	interrupt := cmdutils.InterruptChan()
+
+	cancellableContext, cancel := context.WithTimeout(ctx.Request().Context(), m.probeTimeout)
+
+	defer cancel()
+	var cleanup func() error
+
+	cf := clientconfig.ClientConfigFile{
+		Token:     token,
+		TenantId:  tenant.ID,
+		Namespace: randomNamespace(),
+		TLS: clientconfig.ClientTLSConfigFile{
+			Base: shared.TLSConfigFile{
+				TLSRootCAFile: os.Getenv("SERVER_TLS_ROOT_CA_FILE"),
+			},
+		},
+	}
+	workflowName := "probe-workflow"
+	var wfrId *string
+	go func() {
+		wfrId, cleanup, err = m.run(cancellableContext, cf, workflowName, events, errorChan)
+		if err != nil {
+			errorChan <- err
+		}
+
+		// check if the context was cancelled in the meantime
+
+		if cancellableContext.Err() != nil {
+			cleanupErr := cleanup()
+			if cleanupErr != nil {
+				m.l.Error().Msgf("error cleaning up probe worker: %s", cleanupErr)
+			}
+		}
+	}()
+
+	defer func() {
+		var cleanupErr error
+		if cleanup != nil {
+			cleanupErr = cleanup()
+		} else {
+			m.l.Warn().Msg("cleanup function was never set for probe")
+		}
+		if cleanupErr != nil {
+			m.l.Error().Msgf("error cleaning up probe worker: %s", err)
+		}
+	}()
+
+	defer func() {
+
+		time.Sleep(250 * time.Millisecond)
+
+		if wfrId != nil && *wfrId != "" {
+
+			wrfRow, err := m.config.APIRepository.WorkflowRun().GetWorkflowRunById(context.Background(), cf.TenantId, *wfrId)
+
+			if err != nil {
+				m.l.Error().Msgf("error getting workflow run: %s", err)
+
+			}
+
+			workflowId := sqlchelpers.UUIDToStr(wrfRow.Workflow.ID)
+
+			_, err = m.config.APIRepository.Workflow().DeleteWorkflow(cf.TenantId, workflowId)
+
+			m.l.Info().Msgf("deleted workflow %s", workflowId)
+			if err != nil {
+				m.l.Error().Msgf("error deleting workflow: %s", err)
+			}
+		} else {
+			m.l.Warn().Msg("workflow run id was never set")
+		}
+
+	}()
+
+	messages := []string{"This is a stream event for step-one", "This is a stream event for step-two"}
+	messageIndex := 0
+
+	for {
```

Comment:
We seem to only be checking for stream events -- I think it would be useful to double-check that the steps are called in the correct order and that the steps have the correct output. It'd probably be fine to just read from the database after the workflow run is done.

Reply:
went down a bit of a rabbit hole pulling them from the DB, I think they are not necessarily in order - went back to doing it via the steps.